packages/flutter/lib/src/widgets/default_text_editing_actions.dart

```diff
@@ -245,3 +291,37 @@ class _MoveSelectionToStartTextAction extends TextEditingAction<MoveSelectionToS
     textEditingActionTarget!.renderEditable.moveSelectionToStart(SelectionChangedCause.keyboard);
   }
 }
+
+class _SelectLastTapDownPositionOrEdgeTextAction extends CallbackAction<SingleTapUpTextIntent> {
```

Comment:
These are NOT TextEditingActions like the other Actions in this file.  The reason is that they don't need the machinery in TextEditingAction that checks for a focused TextEditingActionTarget (in fact, they wouldn't work with that, because they need to receive taps on unfocused fields too).  They receive the relevant EditableTextState via the Intent.